Modified to apply to 1.10.3.

commit 3fbdcd0965180b46c545187e7784350340ae88ee
Author: Greg Hudson <ghudson@mit.edu>
Date:   Fri Apr 12 16:28:14 2013 -0400

    Set msg_type when decoding FAST requests
    
    An RFC 6113 KrbFastReq contains a padata sequence and a KDC-REQ-BODY,
    neither of which contain the msg-type field found in a KDC-REQ.  So
    when we decode the FAST request, the resulting krb5_kdc_req structure
    has a msg_type of 0.  Copy msg_type from the outer body, since we make
    use of it in further KDC processing.
    
    ticket: 7605 (new)
    target_version: 1.11.3
    tags: pullup

diff --git a/src/kdc/fast_util.c b/src/kdc/fast_util.c
index 40c5783..4fa36c6 100644
--- a/src/kdc/fast_util.c
+++ b/src/kdc/fast_util.c
@@ -239,6 +239,7 @@ kdc_find_fast(krb5_kdc_req **requestptr,
                                          KRB5_PADATA_FX_COOKIE);
         if (retval == 0) {
             state->fast_options = fast_req->fast_options;
+            fast_req->req_body->msg_type = request->msg_type;
             krb5_free_kdc_req( kdc_context, request);
             *requestptr = fast_req->req_body;
             fast_req->req_body = NULL;
